Source/ARTPushActivationState.m

```diff
@@ -74,7 +74,7 @@ @implementation ARTPushActivationPersistentState
 
 ARTPushActivationState *validateAndSync(ARTPushActivationStateMachine *machine, ARTPushActivationEvent *event) {
     #if TARGET_OS_IOS
-    ARTLocalDevice *const local = machine.rest.device_nosync;
+    ARTLocalDevice *const local = machine.localDevice;
```

Comment:
You have replaced some usages of `rest.device_nosync` with `localDevice`, but there are other uses. If we do this change, we should not leave half the codebase using the old way?
![Screenshot 2021-11-29 at 08 39 37](https://user-images.githubusercontent.com/24711048/143834862-9d5cd255-0243-4e83-bfc1-0130c7335bb2.png)

Reply:
Then, you can possibly remove `device_nosync` and `device` properties from `ARTRestInternal`

https://github.com/ably/ably-cocoa/blob/44cbbb0d18af938411d8787e8af9301e80242813/Source/ARTRest%2BPrivate.h#L20-L23

Reply:
Actually, `localDevice` is a method... Please start with my conversation about the `localDevice` method instead

I will resolve this for now.